pages/1_orders_and_revenue.py

```diff
@@ -18,49 +18,163 @@
+
+# Calculate Created Date
+created_date = data.groupby('customer_id')['customer_created_at'].min().reset_index()
+
+# Merge all the calculated fields
+customer_table = data[['customer_id', 'customer_name', 'customer_email', 'customer_city', 'customer_country']].drop_duplicates().reset_index(drop=True)
+customer_table = customer_table.merge(total_spend, on='customer_id', how='left')
+customer_table = customer_table.merge(total_orders, on='customer_id', how='left')
+customer_table = customer_table.merge(total_refunds, on='customer_id', how='left')
+customer_table = customer_table.merge(last_order_date, on='customer_id', how='left')
+customer_table = customer_table.merge(created_date, on='customer_id', how='left')
+
+# Rename columns
+customer_table.rename(columns={
+    'customer_id': 'ID',
+    'customer_name': 'Name',
+    'customer_email': 'Email',
+    'customer_city': 'City',
+    'customer_country': 'Country',
+    'total_amount': 'Total Spend',
+    'refund_amount': 'Total Refunds',
+    'created_at': 'Last Order Date',
+    'customer_created_at': 'Created Date',
+}, inplace=True)
+
+# Format dollar columns
+customer_table['Total Spend'] = customer_table['Total Spend'].apply(lambda x: f"${x:,.2f}")
+customer_table['Total Refunds'] = customer_table['Total Refunds'].apply(lambda x: f"${x:,.2f}")
+
+# Format date columns
+customer_table['Last Order Date'] = customer_table['Last Order Date'].dt.strftime('%Y-%m-%d')
+customer_table['Created Date'] = customer_table['Created Date'].dt.strftime('%Y-%m-%d')
+
+# Display the customer table in Streamlit
+st.markdown("**Enhanced Customer Table**")
+st.dataframe(customer_table)
+
+
+# Location Performance Chart
+# Aggregate revenue by customer_country
+location_performance = data.groupby('customer_country')['total_amount'].sum().reset_index()
+
+# Sort by total_amount in descending order
+location_performance = location_performance.sort_values(by='total_amount', ascending=False)
+
+# Create the Plotly Express choropleth map
+fig = px.choropleth(
+    location_performance,
+    locations='customer_country',
+    locationmode='country names',
+    color='total_amount',
+    color_continuous_scale=px.colors.sequential.Blues,
+    title='Revenue by Country',
+    labels={'customer_country': 'Country', 'total_amount': 'Total Revenue'},
+    hover_name='customer_country',
+    hover_data={'total_amount': True}
+)
 
-## Location Performance Chart
-st.markdown("**Location Performance**")
+# Update the color bar to show values in thousands
+fig.update_coloraxes(colorbar_tickprefix='$', colorbar_tickformat='~s')
+
+# Update layout to make the map bigger
+fig.update_layout(
+    autosize=False,
+    width=1200,  # Width in pixels
+    height=900,  # Height in pixels
+    title=dict(
+        x=0.5,  # Center title horizontally
+        xanchor='center'
+    )
+)
 
-## Customer Table
-st.markdown("**Customer Table**")
\ No newline at end of file
+# Display the map
+st.plotly_chart(fig, use_container_width=True)
```

Comment:
Let's move this above the customer table.

Reply:
switched!